src/ReferenceList.ts

```diff
@@ -44,8 +44,7 @@ export class ReferenceList {
     }
 
     public translate(movement: number): ReferenceList {
-        assert(movement !== null, "Cannot translate a reference list by a null amount.");
-        assert(movement !== undefined, "Cannot translate a reference list by an undefined amount.");
+        assert(movement);
```

Comment:
>movement [](start = 15, length = 8)

Why do we not need !isNullOrUndefined here?

Reply:
Good catch, can't use a simple falsy check because it could theoretically be zero.  That's why I added isNullOrUndefined, to make these cases easier. (Have I mentioned that JavaScript is a horribly-designed language?).  Now using nonNullValue.